### ISSUE TITLE
Add unicode regex compatibility properties

### ISSUE THREAD

**Issue by kipcole9** [Qqwy/elixir-unicode#4]
@Qqwy seems we overlap in interests more often than not!  In order to bring currency formatting fully up to CLDR spec in my [ex_clcr](https://hex.pm/packages/ex_cldr) package I need to add support for unicode regex properties that underpin `currencySpacing`.  As an example:

```json
          "currencySpacing": {
            "beforeCurrency": {
              "currencyMatch": "[:^S:]",
              "surroundingMatch": "[:digit:]",
              "insertBetween": " "
            },
            "afterCurrency": {
              "currencyMatch": "[:^S:]",
              "surroundingMatch": "[:digit:]",
              "insertBetween": " "
            }
```

Since Erlang's `re` module doesn't support `[:S:]` and friends as you know.  I need to implement that and i would seem reasonable to suggest adding it to your work rather than start anew.  And If I'm doing that then of course I'd add the full character classification for all the compatibility properties.

Would you be receptive to a PR to add character property classification?  I order to avoid multiple passes over the source file I'd probably have to do a bit of refactoring on the code that generates the lookup functions at compile time too.

Perhaps it would seem a reasonable time to update to Unicode 10 as well.

Lastly, I have an emerging need for script identification and therefore wonder if thats in scope of your objectives for this package? This would be based upon unicodes `Scripts.txt` codepoint classification.

In short the proposal is:
1. Update to Unicode 10
2. Add Unicode regex compatibility property functions
3. Add script identification based upon code point(s)

Let me know your thoughts, --Kip

**Comment by Qqwy** [comment]
Great ideas. PRs are very welcome, because I've spread my attention a little thin at the moment :sweat_smile:.
